### PR TITLE
fix service-account related doc

### PR DESCRIPTION
**Pull request by dalanlan**

### CODE REVIEW

File: docs/user-guide/service-accounts.md [hunk 1]
```diff
@@ -120,10 +120,41 @@ $ kubectl delete serviceaccount/build-robot
...
+Annotations:  kubernetes.io/service-account.name=build-robot,kubernetes.io/service-account.uid=870ef2a5-35cf-11e5-8d06-005056b45392
+
+Type: kubernetes.io/service-account-token
+
+Data
+====
+```
 
 TODO explain:
   - The token goes to: "/var/run/secrets/kubernetes.io/serviceaccount/$WHATFILENAME"
```
Comment (dalanlan) [inline docs/user-guide/service-accounts.md]: To be honest i'm quite confused here. According to this [comment](https://github.com/GoogleCloudPlatform/kubernetes/issues/12012#issuecomment-126395595), the manually created secrets wont be mounted to a pod generally. So i cannot see why  we have the mounted volume here.

**Comment by dalanlan** [comment]
@erictune  Spare me a moment to review this, if you please ;-)

**Comment by dalanlan** [comment]
I was wondering if you are able to find time to review this doc related  to SA @liggitt

File: docs/admin/authorization.md [hunk 2]
```diff
@@ -72,13 +73,15 @@ to assist in policy management.
 
 ### Policy File Format
 
-For mode `ABAC`, also specify `--authorization_policy_file=SOME_FILENAME`.
+For mode `ABAC`, also specify `--authorization-policy-file=SOME_FILENAME`.
```
Comment (liggitt) [inline docs/admin/authorization.md]: either works, lots of scripts use `_`, probably not worth changing

File: docs/admin/service-accounts-admin.md [hunk 3]
```diff
@@ -96,13 +96,14 @@ account, and the controller will update it with a generated token:
 ```json
 secret.json:
 {
+  "apiVersion": "v1",
```
Comment (liggitt) [inline docs/admin/service-accounts-admin.md]: indent whitespace

File: docs/admin/authorization.md [hunk 4]
```diff
@@ -72,13 +73,15 @@ to assist in policy management.
...
 
 The file format is [one JSON object per line](http://jsonlines.org/).  There should be no enclosing list or map, just
 one map per line.
 
 Each line is a "policy object".  A policy object is a map with the following properties:
-  - `user`, type string; the user-string from `--token_auth_file`
+  - `user`, type string; the user-string from `--token-auth-file`
+  - `group`, type string; it can be empty if the authenticated user is not in any groups,
```
Comment (liggitt) [inline docs/admin/authorization.md]: that's not quite right... 
* if you specify `user`, it must match the username of the authenticated user
* if you specify `group`, it must match one of the groups of the authenticated user

File: docs/admin/authorization.md [hunk 5]
```diff
@@ -92,7 +95,7 @@ interface.
 
 ### Authorization Algorithm
 
-A request has attributes which correspond to the properties of a policy object.
+A request has attributes which corresponds to the properties of a policy object.
```
Comment (liggitt) [inline docs/admin/authorization.md]: attributes correspond

File: docs/admin/service-accounts-admin.md [hunk 6]
```diff
@@ -118,6 +119,11 @@ kubectl describe secret mysecretname
 kubectl delete secret mysecretname
 ```
 
+### Service Account Controller
+
+ServiceAccountsController is aimed to manage ServiceAccount inside Namespaces, to ensure
```
Comment (liggitt) [inline docs/admin/service-accounts-admin.md]: s/is aimed to manage/manages/

Comment (liggitt) [inline docs/admin/service-accounts-admin.md]: s/to ensure/and ensures/

File: docs/user-guide/service-accounts.md [hunk 7]
```diff
@@ -120,10 +120,41 @@ $ kubectl delete serviceaccount/build-robot
...
+  name: build-robot-secret
+  annotations: 
+    kubernetes.io/service-account.name: build-robot
+type: kubernetes.io/service-account-token
+EOF
+$ kubectl create -f /tmp/build-robot-secret.yaml
+secrets/build-robot-secret
+```
+
+Now you can confirm that the newly built secret is consumed by "build-robot" service account.
```
Comment (liggitt) [inline docs/user-guide/service-accounts.md]: s/is consumed by/is populated with an API token for the/

File: docs/user-guide/service-accounts.md (hunk 1 above)
Comment (liggitt) [inline docs/user-guide/service-accounts.md]: Manually created tokens are typically for manual use outside pods (like an external build/CI/monitoring system). The `/var/run/secrets/...` bit shouldn't be under the manual token section

Comment (dalanlan) [inline docs/user-guide/service-accounts.md]: So i guess this "TODO" part could be removed elegantly?

File: docs/admin/authorization.md (hunk 2 above)
Comment (dalanlan) [inline docs/admin/authorization.md]: Sure:)

File: docs/admin/authorization.md (hunk 5 above)
Comment (dalanlan) [inline docs/admin/authorization.md]: My bad :(

**Comment by dalanlan** [comment]
Updated.

File: docs/user-guide/service-accounts.md [hunk 8]
```diff
@@ -120,14 +120,41 @@ $ kubectl delete serviceaccount/build-robot
...
+Name:   build-robot-secret
+Namespace:  default
+Labels:   <none>
+Annotations:  kubernetes.io/service-account.name=build-robot,kubernetes.io/service-account.uid=870ef2a5-35cf-11e5-8d06-005056b45392
+
+Type: kubernetes.io/service-account-token
+
+Data
+====
+```
```
Comment (liggitt) [inline docs/user-guide/service-accounts.md]: there should be a `token` key in the `Data` section... just elide the content for this example

File: docs/user-guide/service-accounts.md [hunk 9]
```diff
@@ -120,14 +120,41 @@ $ kubectl delete serviceaccount/build-robot
 ```
 
 <!-- TODO: describe how to create a pod with no Service Account. -->
+Note that if a pod does not have a `ServiceAccount` set, the `ServiceAccount` will be set to `default`.
 
 ## Adding Secrets to a service account.
```
Comment (liggitt) [inline docs/user-guide/service-accounts.md]: The doc you've added here should be in a section named something like "Manually create a service account API token"

The TODO about "Test and explain how to use additional non-K8s secrets with an existing service account." still applies, and should describe how to do things like create ssh private key secrets, or docker image pull secrets, and add them to a service account. That can be part of a different PR, but don't remove the TODO yet

**Comment by erictune** [comment]
will lgtm when liggitt gives lgtm

**Comment by erictune** [comment]
@liggitt I made you assignable and assigned you.  Yay.

**Comment by dalanlan** [comment]
Fixed :)

**Comment by liggitt** [comment]
lgtm, thanks

**Comment by dalanlan** [comment]
Rebased again.
Is there any possibility to get this patch merged?  Since it's been quite a while of "lgtm" ;-)

**Comment by liggitt** [comment]
@erictune?

**Comment by dalanlan** [comment]
@liggitt Or could you please kindly add a label "lgtm" :-)

**Comment by liggitt** [comment]
sadly, I don't have label-adding rights :)

File: docs/admin/service-accounts-admin.md [hunk 10]
```diff
@@ -96,13 +96,14 @@ account, and the controller will update it with a generated token:
 ```json
 secret.json:
 {
+    "apiVersion": "v1",
```
Comment (liggitt) [inline docs/admin/service-accounts-admin.md]: line up whitespace

Comment (dalanlan) [inline docs/admin/service-accounts-admin.md]: Yeah, it's kinda odd. 
Seems all right directly from the [page](https://github.com/ZJU-SEL/kubernetes/blob/fix-serviceaccount-doc/docs/admin/service-accounts-admin.md#to-create-additional-api-tokens) via "view" button.
Could it have anything to do with the format here?

Comment (dalanlan) [inline docs/admin/service-accounts-admin.md]: You know what's more weird, `liggitt was unassigned by dalanlan 8 hours ago`.
I was totally offline that time and i dont have the rights without doubt. Sorry for the trouble.

Comment (liggitt) [inline docs/admin/service-accounts-admin.md]: tabs vs spaces... the rest of the example uses tabs

File: docs/admin/service-accounts-admin.md [hunk 11]
```diff
@@ -96,14 +96,15 @@ account, and the controller will update it with a generated token:
...
-	"type": "kubernetes.io/service-account-token"
+    "kind": "Secret",
+    "apiVersion": "v1",
+    "metadata": {
+        "name": "mysecretname",
+        "annotations": {
+            "kubernetes.io/service-account.name": "myserviceaccount"
+        }
+    },
+    "type": "kubernetes.io/service-account-token"
```
Comment (dalanlan) [inline docs/admin/service-accounts-admin.md]: I did try tabs earlier, turns out it didnt fit much.
And now it makes more sense to me after validated by a json validator.
Would you plz kindly have a double check ;-) @liggitt

**Comment by liggitt** [comment]
LGTM, thanks

**Comment by dalanlan** [comment]
@erictune  PTAL

**Pull request merged**